service/src/main/java/tennisclub/service/UserServiceImpl.java

```diff
@@ -0,0 +1,77 @@
+package tennisclub.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import tennisclub.dao.UserDao;
+import tennisclub.entity.User;
+import tennisclub.entity.enums.Role;
+
+import java.util.List;
+
+public class UserServiceImpl implements UserService {
```

Comment:
You are missing `@Service` and `@Transactional` annotations there 😁.

Reply:
Good catch, thanks!